doc/source/index.rst

```diff
@@ -147,7 +147,14 @@ produces:
 
     print('this code is invisible')
 
-or vice versa with ``:hide-output:``::
+this option is particularly useful if you want to embed correctness checks in building your documentation::
+
+  .. jupyter-execute::
+      :hide-code:
+
+      assert everything_works, "There's a bug somewhere"
+
```

Comment:
I'd add something like "even though the code won't make it into your documentation, your documentation build will error if running the code fails."